source/s3_list_objects.c

```diff
@@ -142,7 +142,7 @@ static bool s_on_list_bucket_result_node_encountered(
         }
 
         if (ret_val && operation_data->on_object) {
-            ret_val |= operation_data->on_object(&fs_wrapper.fs_info, operation_data->user_data);
+            ret_val &= operation_data->on_object(&fs_wrapper.fs_info, operation_data->user_data);
```

Comment:
Have you looked into our other uses of the XML parser, in our real code, to see if we're "properly" using `user_data` to report errors?